From abee08bd69752a0528d7883847491df066f2d02c Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Tue, 6 Sep 2022 14:51:27 +0800
Subject: [PATCH] pipe: fix kabi for poll_usage in struct pipe_inode_info
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: abee08bd69752a0528d7883847491df066f2d02c
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5L6BR

--------------------------------

As cd720fad8b57 ("pipe: make poll_usage boolean and annotate its access")
makes poll_usage boolean, kabi compatibility is broken for the interfaces
using struct pipo_inode_info.

Use KABI_REPLACE() to restore kabi compatibility.

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: linan <linan122@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/pipe_fs_i.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/pipe_fs_i.h b/include/linux/pipe_fs_i.h
index c0b6ec6bf65b..b5b380fe264d 100644
--- a/include/linux/pipe_fs_i.h
+++ b/include/linux/pipe_fs_i.h
@@ -71,7 +71,7 @@ struct pipe_inode_info {
 	unsigned int files;
 	unsigned int r_counter;
 	unsigned int w_counter;
-	bool poll_usage;
+	KABI_REPLACE(unsigned int poll_usage, bool poll_usage)
 	struct page *tmp_page;
 	struct fasync_struct *fasync_readers;
 	struct fasync_struct *fasync_writers;
-- 
2.34.1

